run.sh

```diff
@@ -1,33 +1,57 @@
 #!/bin/bash
 set -e
-cd `dirname $0`
-VENV_NAME=viam-venv
+
+UNAME=$(uname -s)
+
+VENV_NAME=".venv"
 echo "Current Directory: $(pwd)"
 echo "Checking for virtual environment folder..."
 
-if [ -d "viam-env" ]
-  then
-    echo "Virtual environment found, activating..."
-    source viam-env/bin/activate
-    echo "Virtual environment activated: $VENV_NAME"
+if [ -d "$VENV_NAME" ]; then
+  echo "Virtual environment found, activating..."
+  source "$VENV_NAME/bin/activate"
+
+  # Check if 'uv' can be imported in Python
+  if ! python3 -c "import uv" > /dev/null 2>&1; then
+    echo "'uv' not found in virtual environment. Recreating the virtual environment..."
+    deactivate
+    rm -rf "$VENV_NAME"
   else
-    echo "Setting up virtual environment..."
-    python3 -m venv --system-site-packages "$VENV_NAME"
-    source "$VENV_NAME/bin/activate"
-    echo "Virtual environment activated: $VENV_NAME"
-    echo "Installing dependencies from requirements.txt..."
+    echo "'uv' is installed, skipping recreation of virtual environment."
+  fi
+fi
 
-    while IFS= read -r requirement; do
-      if [ -n "$requirement" ]; then
-        echo "Installing $requirement..."
-        pip install "$requirement" || echo "Failed to install $requirement"
-      fi
-    done < requirements.txt
+# Create a new virtual environment if it doesn't exist or was removed
+if [ ! -d "$VENV_NAME" ]; then
+  echo "Setting up virtual environment..."
+  
+  if [ "$UNAME" = "Linux" ]; then
+    echo "Installing uv on Linux"
 
-    echo "Dependencies installation complete."
+    # Check if pip is installed
+    if ! command -v pip &> /dev/null; then
+      echo "'pip' not found. Installing pip..."
+      sudo apt-get install -y python3-pip
+    fi
 
+    pip install uv
```

Comment:
Would `pip intall uv` also work on Darwin?

Reply:
pip install uv would work on Darwin if they were already inside a virtual env with python but i didn't want to create nested virtual environments so i went with brew instead.